### ISSUE TITLE
Dictating the word "point" removes spaces before and after the word 

### ISSUE THREAD

**Issue by MarkRx** [dictation-toolbox/dragonfly#311]
**Describe the bug**
When dictating the word "point" using Caster spaces are removed before and after it. The Dragon dictation popup shows the correct words. When using Dragon in dictation it works correctly.

It's because the word "point" is treated as special at [dictation_format.py 412](https://github.com/dictation-toolbox/dragonfly/blame/master/dragonfly/engines/backend_natlink/dictation_format.py#L412) and [dictation_format.py 339](https://github.com/dictation-toolbox/dragonfly/blame/master/dragonfly/engines/backend_natlink/dictation_format.py#L339).

```
Parsed input u'what' -> Word(u'what')
Parsed input u'is' -> Word(u'is')
Parsed input u'the' -> Word(u'the')
Parsed input u'point' -> Word(u'point', no_space_before, no_space_after, no_space_between)
Parsed input u'of' -> Word(u'of')
Parsed input u'that' -> Word(u'that')
```

**To Reproduce**
1. Say "cop what is the point of that questo" (if using caster)
2. Output is "What is thepointof that?"

Here are more sentences that don't work:
"I received a letterin the mail"
"I had todashout the door"
(any word that is hard coded may misbehave)

**Expected behavior**
Output is "What is the point of that?"

**System:**
OS: Windows 10
Program: Any
Version: Caster master 3f45fbb, dragonfly 0.27.1
Dictation Engine: Dragon Pro 15

**Comment by MarkRx** [comment]
While I don't know the context it seems to me that [dictation_format.py 412](https://github.com/dictation-toolbox/dragonfly/blame/master/dragonfly/engines/backend_natlink/dictation_format.py#L412) should be removed. The color of the dragonfly library should be responsible for special formatting - why does dragonfly hard code special words?

**Comment by LexiconCode** [comment]
The say command simply passes through straight from the engine the formatting. Therefore it is strange to me is there's a different behavior between normal dictation with vanilla DNS/DPI and the dragonfly dictation element which the `say` command utilizes.

> The caller of the dragonfly library should be responsible for special formatting - why does the dragonfly DNS 11+ engine hard code special words that double up as formatting properties?

While I could attempt to answer that I believe @Danesprite could do a much better job with a concise answer.

**Comment by Danesprite** [comment]
Thanks for reporting this @MarxRx. The reason why "point" is formatted like this is because the word can also be recognised as ".\point\point". This is used for saying numbers. For example, if you say "one point two three four", Dragon will recognise that as "one\number .\point\point two three four". The default behaviour will be to type 1.234.

Clearly, in this case it makes sense to remove the spaces. Dragonfly's DNS formatting code needs to be adjusted to tell the difference between ".\point\point" and "point".

**Comment by Danesprite** [comment]
This bug was introduced by PR #111.

**Comment by Danesprite** [comment]
@MarkRx This is now fixed in version 0.28.1. You can upgrade to that version by running the following command:

```Shell
pip install --upgrade dragonfly2
```
